pkg/provisioner/provisioner.go

```diff
@@ -72,6 +72,10 @@ type Provisioner interface {
 	// PowerOff ensures the server is powered off independently of any image
 	// provisioning operation.
 	PowerOff() (result Result, err error)
+
+	// SoftPowerOff tries to perform soft shutdown first
+	// If soft shutdown is unspported, it performs hard shutdown
+	SoftPowerOff() (result Result, err error)
```

Comment:
Do we need to add a new method? Could we change the implementation of PowerOff() instead?